fpdcc/ccfp-asset-dashboard
# Add models for the GIS database

## Overview
This PR adds the models for the `quercus` schema. For #61.

Includes:
- Unmanaged models with the relationships.
  - I followed what I could observe in my local database. See the schema from postgres for quick reference (see `postgis_schema.md`).
  - I added docstrings from [their documentation](https://github.com/fpdcc/triggers_and_updates/blob/master/documentation.md).
- Tests for the models and relationships.
  - I added a new docker command to the README so we can test the model relationships when the tests run in a local development environment.
- Pinned requirements after issues with Django 4.0. I copy/pasted the requirements in PR #82. 

### Notes
I wasn't sure how to use the local `fp-postgis` database without adding a boolean flag to the docker-compose command. I don't know if this is anti-pattern or what -- couldn't figure out how to combine the existing code in `conftest.py` with the local `fp-postgis` docker container. I thought creating test fixtures would be unnecessary work since we already have the data we need in that container's database. These tests won't run in a CI/CD action. They're meant to run only in a developer's local environment, with the `fp-postgis` database. Let me know if there's a better way to do this (cc @hancush)

Any ideas on how to make the tests more DRY?

Should we move these models to their own file? If so, what would be a good name?

## Testing Instructions
- Pull down this branch `git pull origin unmanaged-models && git checkout unmanaged-models`
- Run the tests: `docker-compose -f docker-compose.yml -f tests/docker-compose.yml run -e TEST_GIS=True --rm app`.
  - Make sure that you have the `fp-postgis` database loaded (see README instructions if not).
- Make sure the models are implemented the right way based on the schema located at `postgis_schema.md`.
